### ISSUE TITLE
Use `age` or `OpenPGP` instead of password

### ISSUE THREAD

**Issue by radumarias** [xoriors/rencfs#233]
How password works
- From the password, we generate a key with KDF. Let's call it `KEK`
- We generate a random master encryption key, let' call it `DEK` used to encrypt the files
- We encrypt the master encryption key with `KEK`

How we'll change it instead of using the password

The first time we run the app
- We set age or PGP, if not set already
- For that, we will use an age or PGP client
- We generate a random master encryption key used to encrypt the files
- We encrypt the master encryption key with `KEK`
- We encrypt `KEK` with age

When open it again
- Decrypt `KEK` with age or PGP
- Decrypt `DEK` with `KEK`

https://docs.rs/age/0.11.1/age/#recipient-based-encryption

**Comment by BLRGAdityaC** [comment]
Hi there @radumarias, I recently contacted you on Slack regarding working on this Issue. I will contact you on Slack or post relevant questions/clarifications here as I work on the project.

**Comment by radumarias** [comment]
> Hi there @radumarias, I recently contacted you on Slack regarding working on this Issue. I will contact you on Slack or post relevant questions/clarifications here as I work on the project.

sure

**Comment by radumarias** [comment]
@BLRGAdityaC how's it going with this? is there any help needed from my side?

**Comment by BLRGAdityaC** [comment]
@radumarias You had briefly mentioned on Slack that you wanted to create an abstraction layer before I could continue work on this issue. Is that still the case?

**Comment by radumarias** [comment]
yes, I started but haven't managed to finish it as I was caught up with my full time job
but I'll try to continue over the next day, will let you know

**Comment by BLRGAdityaC** [comment]
Of course, let me know if you change your mind, I'm willing to try and make my own solution work.

**Comment by radumarias** [comment]
@BLRGAdityaC please give it try to your solution. I'm sorry but I'm very caught up and cannot do that abstraction I said

**Comment by radumarias** [comment]
@BLRGAdityaC how are you with this?

**Comment by BLRGAdityaC** [comment]
Unfortunately, it seems as though I cannot accurately predict how long I'll be working on the issue given personal circumstances. As such, I will un-assign myself from this issue to allow someone else to work on it. I will occasionally continue working on this issue in my own time within the coming months and may reassign myself to it if no one is willing to work on it after a couple months.

**Comment by radumarias** [comment]
thanks, but I think you can keep it on you and work on it as you can, lete
know any questions you have

And in the end, it's not the years in your life that count. It's the life
in your years.
